python/MaterialX/__init__.py

```diff
@@ -1,3 +1,20 @@
+# Python 3.8+ on Windows: DLL search paths for dependent
+# shared libraries
+# Refs.:
+# - https://github.com/python/cpython/issues/80266
+# - https://docs.python.org/3.8/library/os.html#os.add_dll_directory
+import os
+import sys
+if sys.platform == "win32" and sys.version_info >= (3, 8):
+    # On a standard installation, this file is in %INSTALLDIR%\python\MaterialX
+    # We need to add %INSTALLDIR%\bin to the DLL path.
+    mxdir = os.path.dirname(__file__)
+    pydir = os.path.split(mxdir)[0]
+    installdir = os.path.split(pydir)[0]
+    bindir = os.path.join(installdir, "bin")
+    if os.path.exists(bindir):
+        os.add_dll_directory(bindir)
```

Comment:
Once we start shipping MaterialX on PyPI, this could be considered "dangerous" and could cause troubles for MaterialX users.

One thing that could be done is to fence like this:

```python
if sys.platform == "win32" and sys.version_info >= (3, 8):
    try:
        importlib.metadata.version('MaterialX')
    except importlib.metadata.PackageNotFoundError:
        mxdir = os.path.dirname(__file__)
        pydir = os.path.split(mxdir)[0]
        installdir = os.path.split(pydir)[0]
        bindir = os.path.join(installdir, "bin")
        if os.path.exists(bindir):
            os.add_dll_directory(bindir)
```
This would ensure that we don't add "random" stuff to the DLL search directories when installing MaterialX with pip.